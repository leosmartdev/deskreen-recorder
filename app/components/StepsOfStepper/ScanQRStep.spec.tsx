/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react';
import Enzyme from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ScanQRStep from './ScanQRStep';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

const bp3QRCodeDialogRootSelector = '#bp3-qr-code-dialog-root';
const magnifyQRCodeButtonSelector = '#magnify-qr-code-button';
const qrCodeDialogInnerSelector = '#qr-code-dialog-inner';

describe('<ScanQRStep />', () => {
  let wrapper = Enzyme.shallow(<ScanQRStep />);

  beforeEach(() => {
    wrapper = Enzyme.shallow(<ScanQRStep />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('when rendered', () => {
    it('should match exact snapshot', () => {
      expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('when user clicks on magnify QR code button', () => {
    it('should set "QR Code Dialog Root" isOpen property to "true"', () => {
      // @ts-ignore
      wrapper.find(magnifyQRCodeButtonSelector).props().onClick();
      // @ts-ignore
      expect(wrapper.find(bp3QRCodeDialogRootSelector).props().isOpen).toBe(
        true
      );
    });
  });

  describe(`when magnified QR code dialog is opened,
  and when user closes magnified QR code dialog`, () => {
    it('should set "QR Code Dialog Root" isOpen property to "false"', () => {
      // @ts-ignore
      wrapper.find(magnifyQRCodeButtonSelector).props().onClick();
      // @ts-ignore
      wrapper.find(bp3QRCodeDialogRootSelector).props().onClose();
      // @ts-ignore
      expect(wrapper.find(bp3QRCodeDialogRootSelector).props().isOpen).toBe(
        false
      );
    });
  });

  describe(`when magnified QR code dialog is opened,
  and when user clicks on qr code dialog inner`, () => {
    it('should set "QR Code Dialog Root" isOpen property to "false"', () => {
      // @ts-ignore
      wrapper.find(magnifyQRCodeButtonSelector).props().onClick();
      // @ts-ignore
      wrapper.find(qrCodeDialogInnerSelector).props().onClick();
      // @ts-ignore
      expect(wrapper.find(bp3QRCodeDialogRootSelector).props().isOpen).toBe(
        false
      );
    });
  });
});
