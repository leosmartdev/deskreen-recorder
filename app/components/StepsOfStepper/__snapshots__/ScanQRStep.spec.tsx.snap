// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScanQRStep /> when rendered should match exact snapshot 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <Blueprint3.Text
      ellipsize={false}
      tagName="div"
    >
      <span
        style={
          Object {
            "backgroundColor": "#00f99273",
            "borderRadius": "20px",
            "fontWeight": 900,
            "paddingLeft": "8px",
            "paddingRight": "8px",
          }
        }
      />
    </Blueprint3.Text>
    <Blueprint3.Text
      className="bp3-text"
      ellipsize={false}
      tagName="div"
    />
  </div>
  <div>
    <Blueprint3.Tooltip
      content=""
      hoverCloseDelay={0}
      hoverOpenDelay={100}
      minimal={false}
      position="left"
      transitionDuration={100}
    >
      <Blueprint3.Button
        className="makeStyles-smallQRCode-1"
        id="magnify-qr-code-button"
        onClick={[Function]}
      >
        <QRCode
          bgColor="rgba(0,0,0,0.0)"
          fgColor="#000000"
          imageSettings={
            Object {
              "height": 40,
              "src": "http://127.0.0.1:3000/logo192.png",
              "width": 40,
            }
          }
          includeMargin={false}
          level="H"
          renderAs="svg"
          size={128}
          value="http://:3000/"
        />
      </Blueprint3.Button>
    </Blueprint3.Tooltip>
  </div>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <Blueprint3.Text
      className="bp3-text-muted"
      ellipsize={false}
      tagName="div"
    >
      :
    </Blueprint3.Text>
  </div>
  <Blueprint3.Tooltip
    content=""
    hoverCloseDelay={0}
    hoverOpenDelay={100}
    minimal={false}
    position="left"
    transitionDuration={100}
  >
    <Blueprint3.Button
      icon="duplicate"
      intent="primary"
      onClick={[Function]}
      style={
        Object {
          "borderRadius": "100px",
        }
      }
    >
      http://:3000/
    </Blueprint3.Button>
  </Blueprint3.Tooltip>
  <Blueprint3.Dialog
    canEscapeKeyClose={true}
    canOutsideClickClose={true}
    className="makeStyles-bigQRCodeDialogRoot-3"
    id="bp3-qr-code-dialog-root"
    isOpen={false}
    onClose={[Function]}
    style={
      Object {
        "position": "relative",
        "top": "0px",
      }
    }
    transitionDuration={0}
    usePortal={false}
  >
    <Row
      center="xs"
      className="bp3-dialog-body"
      id="qr-code-dialog-inner"
      middle="xs"
      onClick={[Function]}
    >
      <Col
        className="makeStyles-dialogQRWrapper-2"
        xs={11}
      >
        <QRCode
          bgColor="#FFFFFF"
          fgColor="#000000"
          height="390px"
          imageSettings={
            Object {
              "height": 25,
              "src": "http://127.0.0.1:3000/logo192.png",
              "width": 25,
            }
          }
          includeMargin={false}
          level="H"
          renderAs="svg"
          size={128}
          value="http://:3000/"
          width="390px"
        />
      </Col>
    </Row>
  </Blueprint3.Dialog>
</Fragment>
`;
